src/ampJson.js

```diff
@@ -1,13 +1,15 @@
-import { sanitizeAttrVal, sanitizeElementName, sanitizeJSON } from './sanitize';
+import { sanitizeJSON } from './sanitize';
+import createElement from './createElement';
 
 export default function ampJson(data, component, type) {
   if (!data || !component) return '';
 
-  return Object.entries(data).map(([key, value]) => (
-    `<${sanitizeElementName(component)} id="${sanitizeAttrVal(key)}"${type ? ` type="${sanitizeAttrVal(type)}"` : ''}>
-      <script type="application/json">
-      ${sanitizeJSON(value)}
-      </script>
-    </${sanitizeElementName(component)}>`
+  return Object.entries(data).map(([id, value]) => createElement(
+    component,
+    {
+      id,
+      type,
+    },
+    `<script type="application/json">${sanitizeJSON(value)}</script>`,
```

Comment:
should this use `createElement('script', { type: 'application/json' }, value)` as well?